workspaces/ui-v2/src/pages/diffs/ReviewDiffPages.tsx

```diff
@@ -12,28 +14,62 @@ import { SharedDiffStore } from '<src>/pages/diffs/contexts/SharedDiffContext';
 import { PendingEndpointPageSession } from './PendingEndpointPage';
 import { DiffUrlsPage } from './AddEndpointsPage';
 import { ReviewEndpointDiffContainer } from './ReviewEndpointDiffPage';
-import { useDiffsForCapture } from '<src>/pages/diffs/hooks/useDiffForCapture';
-import { v4 as uuidv4 } from 'uuid';
-import { selectors, useAppSelector } from '<src>/store';
+import {
+  diffActions,
+  selectors,
+  useAppDispatch,
+  useAppSelector,
+} from '<src>/store';
 import { useFetchEndpoints } from '<src>/hooks/useFetchEndpoints';
 import {
   CapturePageWithoutDiffOrRedirect,
   CapturePageWithDiff,
 } from './CapturePage';
 import { PageLayout } from '<src>/components';
 import { LoadingDiffReview } from '<src>/pages/diffs/components/LoadingDiffReview';
+import { useCapturesService } from '<src>/hooks/useCapturesHook';
 import { usePaths } from '<src>/hooks/usePathsHook';
 import { IRequestBody } from '<src>/types';
+import { IOpticDiffService } from '../../../../spectacle/build';
+import { useAnalytics } from '<src>/contexts/analytics';
 
 export function DiffReviewPages(props: any) {
   const { match } = props;
   const { boundaryId } = match.params;
-  const diffId = useMemo(() => uuidv4(), []);
+  const analytics = useAnalytics();
+  const capturesService = useCapturesService();
+  const [diffService, setDiffService] = useState<IOpticDiffService | null>(
+    null
+  );
+
+  const dispatch = useAppDispatch();
 
-  //dependencies
-  const diff = useDiffsForCapture(boundaryId, diffId);
   useFetchEndpoints();
   const endpointsState = useAppSelector((state) => state.endpoints.results);
+  useEffect(() => {
+    (async () => {
+      const startTime = Date.now();
+      const actionResult = await dispatch(
+        diffActions.fetchDiffsForCapture({
+          capturesService,
+          captureId: boundaryId,
+        })
+      );
+
+      try {
+        const results = await unwrapResult(actionResult);
+        const endTime = Date.now();
+        setDiffService(results.diffService);
+        analytics.reviewPageLoaded(
+          results.data.diffs.length,
+          results.data.urls.length,
+          endTime - startTime,
+          results.numberOfEndpoints
```

Comment:
This is odd, but it's because we are sending this information on a reviewPageLoaded. It probably should go in as a child and on a first mount. This is also more correct what's currently implemented, where we potentially send multiple reviewPageLoaded events